test/type-resolution/callables.test.ts

```diff
@@ -577,3 +577,133 @@ test('Single arguments are interpreted as the first argument of a call', async (
+        },
+      },
+      Bucket: {
+        Type: 'aws-cdk-lib.aws_s3.Bucket',
+        Properties: {
+          encryptionKey: {
+            Ref: 'Key',
+          },
+        },
+      },
+    },
+  });
+
+  const typedTemplate = new TypedTemplate(template, { typeSystem });
+
+  expect(typedTemplate.resource('Key')).toEqual(
+    expect.objectContaining({
+      call: {
+        args: {
+          array: [
+            {
+              reference: {
+                fn: 'ref',
+                logicalId: 'CDK::Scope',
+                type: 'intrinsic',
+              },
+              type: 'resolve-reference',
+            },
+            {
+              type: 'string',
+              value: 'Key',
+            },
+            {
+              type: 'string',
+              value: 'some-key',
+            },
+          ],
+          type: 'array',
+        },
+        fqn: 'aws-cdk-lib.aws_kms.Key',
+        method: 'fromKeyArn',
+        namespace: 'aws_kms',
+        type: 'staticMethodCall',
+      },
+    })
+  );
+});
+
+test('Scope and id are unwrapped when passed through CDK::Args', async () => {
+  const template = await Template.fromObject({
+    Resources: {
+      Key: {
+        Call: {
+          'aws-cdk-lib.aws_kms.Key.fromKeyArn': {
+            'CDK::Args': [{ Ref: 'CDK::Scope' }, 'Key', 'some-key'],
+          },
+        },
+      },
+      Bucket: {
+        Type: 'aws-cdk-lib.aws_s3.Bucket',
+        Properties: {
+          encryptionKey: {
+            Ref: 'Key',
+          },
+        },
+      },
+    },
+  });
+
+  const typedTemplate = new TypedTemplate(template, { typeSystem });
+
+  expect(typedTemplate.resource('Key')).toEqual(
+    expect.objectContaining({
+      call: {
+        args: {
+          array: [
+            {
+              reference: {
+                fn: 'ref',
+                logicalId: 'CDK::Scope',
+                type: 'intrinsic',
+              },
+              type: 'resolve-reference',
+            },
+            {
+              type: 'string',
+              value: 'Key',
+            },
+            {
+              type: 'string',
+              value: 'some-key',
+            },
+          ],
+          type: 'array',
+        },
+        fqn: 'aws-cdk-lib.aws_kms.Key',
+        method: 'fromKeyArn',
+        namespace: 'aws_kms',
+        type: 'staticMethodCall',
+      },
+    })
+  );
+});
+
+test('Calls to fromXxx methods cannot be inlined', async () => {
+  const template = await Template.fromObject({
+    Resources: {
+      Bucket: {
+        Type: 'aws-cdk-lib.aws_s3.Bucket',
+        Properties: {
+          encryptionKey: {
+            'aws-cdk-lib.aws_kms.Key.fromKeyArn':
+              'arn:aws:kms:us-east-1:111111111111:key/93726116-3886-4976-885d-035f6c630059',
+          },
+        },
+      },
+    },
+  });
+
+  expect(() => new TypedTemplate(template, { typeSystem })).toThrow(
+    /failed because the id could not be inferred/
```

Comment:
I suppose we could try and generate an id even for cases like this. For example, the id could be the logical id of the resource + the property name (`BucketencryptionKey` in this case). But this would be a complex change in itself (and I'm not even sure this can be done for all cases).

This is a one way door decision. Allowing implicit calls later would not be a breaking change.

Reply:
To clarify: Does the explicit version with `CDK:Args` work here?